src/migrations/20251218153722-predefined-project-role-reader.js

```
@@ -0,0 +1,25 @@
+
+exports.up = (db, callback) => {
+    db.runSql(
+        `
+        ALTER TABLE roles DROP CONSTRAINT unique_name;
+        CREATE UNIQUE INDEX roles_type_name_unique ON roles (name, type);
+
+        INSERT INTO roles (name, description, type)
+        VALUES ('Reader', 'Users with the project reader role have read-only access to the project and cannot make changes.', 'project')
+        ON CONFLICT (name, type) DO NOTHING;
```

Should the migration fail on conflict? Otherwise, we may end up in an invalid state